### DISCUSSION TITLE
v0.18.0

### DISCUSSION THREAD

**Discussion by cschleiden** [cschleiden/go-workflows#298] [Announcements]
🎉 A packed release with lots of features,  bugfixes and performance improvements. Big thanks to the new contributors 🙇 

### Breaking Changes / Features

* Use golang-migrate for schema migrations for SQL backends by @cschleiden in https://github.com/cschleiden/go-workflows/pull/261
* Use mediumblob for mysql attributes by @cschleiden in https://github.com/cschleiden/go-workflows/pull/264
* Execute workflows by (custom) name by @cschleiden in https://github.com/cschleiden/go-workflows/pull/294
* Explicitly evict finished workflow instances from cache by @cschleiden in https://github.com/cschleiden/go-workflows/pull/278
* Add pending workflow tasks metric to stats interface by @cschleiden in https://github.com/cschleiden/go-workflows/pull/279
* Prevent multiple active instances with the same id by @cschleiden in https://github.com/cschleiden/go-workflows/pull/288

#### Performance

* Store event payloads in separate key by @cschleiden in https://github.com/cschleiden/go-workflows/pull/273
* Sqlite: store payloads/attributes in separate table by @cschleiden in https://github.com/cschleiden/go-workflows/pull/267
* Move mysql attributes to separate table by @cschleiden in https://github.com/cschleiden/go-workflows/pull/275

### Bugfixes

* Fix activity heartbeat by @cschleiden in https://github.com/cschleiden/go-workflows/pull/269
* Follow up fix for activity task extension by @cschleiden in https://github.com/cschleiden/go-workflows/pull/271
* Ignore polling interval for redis backend by @cschleiden in https://github.com/cschleiden/go-workflows/pull/277
* fix: redis diagnostics panic by @decibelcooper in https://github.com/cschleiden/go-workflows/pull/287
* Use correct key for checking whether an instance exists by @cschleiden in https://github.com/cschleiden/go-workflows/pull/292
* Track newly created instances correctly by @cschleiden in https://github.com/cschleiden/go-workflows/pull/297
* Delete and expire event payloads by @cschleiden in https://github.com/cschleiden/go-workflows/pull/289

### Other Changes

* Unify worker code by @cschleiden in https://github.com/cschleiden/go-workflows/pull/274
* Use HASH for event payloads by @cschleiden in https://github.com/cschleiden/go-workflows/pull/286

* Run activities in parallel for `concurrent` sample by @cschleiden in https://github.com/cschleiden/go-workflows/pull/257
* Enable large benchmark for workflow by @cschleiden in https://github.com/cschleiden/go-workflows/pull/266
* Upgrade test-summary action by @cschleiden in https://github.com/cschleiden/go-workflows/pull/276
* Fix flags for scale sample by @cschleiden in https://github.com/cschleiden/go-workflows/pull/281
* Fix some linting errors by @cschleiden in https://github.com/cschleiden/go-workflows/pull/290
* Refactor workflow instance creation by @cschleiden in https://github.com/cschleiden/go-workflows/pull/293

## New Contributors
* @decibelcooper made their first contribution in https://github.com/cschleiden/go-workflows/pull/287

**Full Changelog**: https://github.com/cschleiden/go-workflows/compare/v0.17.0...v0.18.0

<hr /><em>This discussion was created from the release <a href='https://github.com/cschleiden/go-workflows/releases/tag/v0.18.0'>v0.18.0</a>.</em>

**Comment by dewrich** [comment]
@cschleiden I love where this project is going.  Any thoughts on when this might become more "official", instead of beta?  I'd like to start taking a closer look for production use cases.

**Reply by cschleiden** [reply]
@dewrich There is a tracking issue for moving to a 1.0. As of now, there's not much left: #247
